File: PYTHON/PetShop.py
class PetShop:
    
    def __init__(self, id="", nama_produk="", harga_produk=0, stok_produk=0):
        # Konstruktor untuk menginisialisasi atribut dengan nilai default
        self.set_id(id)
        self.set_nama_produk(nama_produk)
        self.set_harga_produk(harga_produk)
        self.set_stok_produk(stok_produk)

    # Mengeset nilai atribut ID
    def set_id(self, id):
        self._id = id  # Menyimpan nilai ID ke dalam atribut privat
    
    # Mengembalikan nilai atribut ID
    def get_id(self):
        return self._id  # Mengembalikan nilai ID

    # Mengeset nilai atribut nama produk
    def set_nama_produk(self, nama_produk):
        self._nama_produk = nama_produk  # Menyimpan nama produk ke dalam atribut privat
    
    # Mengembalikan nilai atribut nama produk
    def get_nama_produk(self):
        return self._nama_produk  # Mengembalikan nama produk
    
    # Mengeset nilai atribut harga produk
    def set_harga_produk(self, harga_produk):
        if harga_produk >= 0:
            self._harga_produk = harga_produk  # Menyimpan harga produk jika tidak negatif
        else:
            raise ValueError("Harga produk tidak boleh negatif.")  # Menampilkan error jika harga negatif
    
    # Mengembalikan nilai atribut harga produk
    def get_harga_produk(self):
        return self._harga_produk  # Mengembalikan harga produk
    
    # Mengeset nilai atribut stok produk
    def set_stok_produk(self, stok_produk):
        self._stok_produk = stok_produk  # Menyimpan stok produk ke dalam atribut privat
    
    # Mengembalikan nilai atribut stok produk
    def get_stok_produk(self):
        return self._stok_produk  # Mengembalikan stok produk
    
    def display(self, id_max, nama_max, harga_max, stok_max):
        # Menampilkan data produk dengan format yang rapi berdasarkan panjang maksimal kolom
        print(
            self._id + " " * (id_max - len(self._id)),
            self._nama_produk + " " * (nama_max - len(self._nama_produk)),
            str(self._harga_produk) + " " * (harga_max - len(str(self._harga_produk))),
            str(self._stok_produk) + " " * (stok_max - len(str(self._stok_produk)))
        )

File: PHP/PetShop.php
<?php
// PetShop.php

// Mendefinisikan kelas PetShop
class PetShop {
    // Mendeklarasikan properti yang digunakan dalam kelas
    protected $id;
    protected $nama_produk;
    protected $harga_produk;
    protected $stok_produk;
    protected $foto;

    // Konstruktor default untuk menginisialisasi properti
    public function __construct($id = "", $nama_produk = "", $harga_produk = 0, $stok_produk = 0, $foto = "") {
        $this->id = $id;
        $this->nama_produk = $nama_produk;
        $this->harga_produk = $harga_produk;
        $this->stok_produk = $stok_produk;
        $this->foto = $foto;
    }

    // Metode untuk mengatur nilai ID
    public function setId($id) {
        $this->id = $id;
    }
    
    // Metode untuk mendapatkan nilai ID
    public function getId() {
        return $this->id;
    }

    // Metode untuk mengatur nama produk
    public function setNamaProduk($nama_produk) {
        $this->nama_produk = $nama_produk;
    }
    
    // Metode untuk mendapatkan nama produk
    public function getNamaProduk() {
        return $this->nama_produk;
    }

    // Metode untuk mengatur harga produk
    public function setHargaProduk($harga_produk) {
        $this->harga_produk = $harga_produk;
    }

    // Metode untuk mendapatkan harga produk
    public function getHargaProduk() {
        return $this->harga_produk;
    }

    // Metode untuk mengatur stok produk
    public function setStokProduk($stok_produk) {
        $this->stok_produk = $stok_produk;
    }

    // Metode untuk mendapatkan stok produk
    public function getStokProduk() {
        return $this->stok_produk;
    }

    // Metode untuk mengatur foto produk
    public function setFoto($foto) {
        $this->foto = $foto;
    }
    
    // Metode untuk mendapatkan foto produk
    public function getFoto() {
        return $this->foto;
    }

    // Metode untuk menampilkan informasi produk dalam format yang terstruktur
    public function display($id_max, $nama_max, $harga_max, $stok_max, $foto_max = null) {
        // Menampilkan ID produk dan menyesuaikan lebar tampilan
        echo $this->id . str_repeat(" ", $id_max - strlen($this->id))
            . $this->nama_produk . str_repeat(" ", $nama_max - strlen($this->nama_produk))
            . $this->harga_produk . str_repeat(" ", $harga_max - strlen((string)$this->harga_produk))
            . $this->stok_produk . str_repeat(" ", $stok_max - strlen((string)$this->stok_produk));
        
        // Jika foto_max disediakan, maka akan menampilkan foto produk
        if ($foto_max !== null) {
            echo $this->foto . str_repeat(" ", $foto_max - strlen($this->foto));
        }
    }
}
?>
